Dalamud/Game/Gui/Nameplates/NameplateGui.cs

```diff
@@ -0,0 +1,177 @@
+    private NameplateGui(TargetSigScanner sigScanner)
+    {
+        // Services
+        this.gameGui = Service<GameGui>.Get();
+        this.objectTable = Service<ObjectTable>.Get();
+        this.addonLifecycle = Service<AddonLifecycle>.Get();
+
+        // Pointers
+        this.addonLifecycle.RegisterListener(new AddonLifecycleEventListener(AddonEvent.PostSetup, "NamePlate", this.AddonLifecycle_Event));
+        this.addonLifecycle.RegisterListener(new AddonLifecycleEventListener(AddonEvent.PreFinalize, "NamePlate", this.AddonLifecycle_Event));
+
+        // Address resolver
+        this.hookAddresses = new();
+        this.hookAddresses.Setup(sigScanner);
+
+        // Hooks
+        this.setPlayerNameplateDetourHook = Hook<SetPlayerNameplateDetourDelegate>.FromAddress(this.hookAddresses.SetPlayerNameplateDetour, this.HandleSetPlayerNameplateDetour);
+        this.setPlayerNameplateDetourHook.Enable();
+    }
+
+    private unsafe delegate nint SetPlayerNameplateDetourDelegate(nint playerNameplateObjectPtr, bool isTitleAboveName, bool isTitleVisible, nint titlePtr, nint namePtr, nint freeCompanyPtr, nint prefix, int iconId);
+
+    /// <inheritdoc/>
+    public event INameplatesGui.OnNameplateUpdateDelegate OnNameplateUpdate;
+
+    /// <inheritdoc/>
+    public void DisposeService()
+    {
+        this.setPlayerNameplateDetourHook.Dispose();
+    }
+
+    /// <inheritdoc/>
+    public T? GetNameplateGameObject<T>(INameplateObject nameplateObject) where T : GameObject
+    {
+        return this.GetNameplateGameObject<T>(nameplateObject.Pointer);
+    }
+
+    /// <inheritdoc/>
+    public T? GetNameplateGameObject<T>(nint nameplateObjectPtr) where T : GameObject
+    {
+        // Get the nameplate object array
+        var nameplateAddonPtr = this.gameGui.GetAddonByName("NamePlate", 1);
+        var nameplateObjectArrayPtrPtr = nameplateAddonPtr + Marshal.OffsetOf(typeof(AddonNamePlate), nameof(AddonNamePlate.NamePlateObjectArray)).ToInt32();
+        var nameplateObjectArrayPtr = Marshal.ReadIntPtr(nameplateObjectArrayPtrPtr);
+
+        if (nameplateObjectArrayPtr == nint.Zero)
+            return null;
+
+        // Determine the index of the nameplate object within the nameplate object array
+        var namePlateObjectSize = Marshal.SizeOf(typeof(AddonNamePlate.NamePlateObject));
+        var namePlateObjectPtr0 = nameplateObjectArrayPtr + namePlateObjectSize * 0;
+        var namePlateIndex = (nameplateObjectPtr.ToInt64() - namePlateObjectPtr0.ToInt64()) / namePlateObjectSize;
+        
+        if (namePlateIndex < 0 || namePlateIndex >= AddonNamePlate.NumNamePlateObjects)
+            return null;
+
+        // Get the nameplate info array
+        var nameplateInfoArrayPtr = nint.Zero;
+        unsafe
+        {
+            var framework = FFXIVClientStructs.FFXIV.Client.System.Framework.Framework.Instance();
+            nameplateInfoArrayPtr = new nint(&framework->GetUIModule()->GetRaptureAtkModule()->NamePlateInfoArray);
```

Comment:
Seems like this doesn't work anymore. Any idea?

Reply:
Should still be there. It’s inherited from UIModuleInterface now via the generators. Just build it and maybe restart your IDE.

Reply:
I _think_ I got it now. Thanks! I even could simplify the method `GetNameplateGameObject` a bit more.